tests/buildTools/src/test/scala/tests/GradleBuildToolSuiteBase.scala

```diff
@@ -27,28 +27,25 @@ abstract class GradleBuildToolSuiteBase(val allGradle: List[String])
   def checkGradleBuild(
       title: TestOptions,
       setup: String,
-      gradleVersions: List[String],
       expectedSemanticdbFiles: Int = 0,
       expectedPackages: String = "",
-      extraArguments: List[String] = Nil
+      extraArguments: List[String] = Nil,
+      gradleVersions: List[Tool.Gradle] = Nil,
+      tools: List[Tool] = Nil
```

Comment:
This extra argument allows further specifying compilers/buildtools/etc. that are used as part of the test, which can affect the success under particular JDK.

E.g. if a Gradle build can run under JDK 21, but uses a Scala 2.12.12 compiler, it won't be able to complete because JDK 21 support requires at least Scala 2.12.18